samples/sample-07/src/test/java/com/example/sample07/Sample07ApplicationTest.java

```diff
@@ -0,0 +1,335 @@
+							.collect(Collectors.toMap(
+										s -> s,
+										s -> (String) properties.get(s)));
+		}
+		catch (IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@EnableKafka
+	@TestConfiguration
+	public static class TestConfig {
+
+		CountDownLatch partitionRevokedLatch = new CountDownLatch(0);
+
+		CountDownLatch partitionAssignedLatch = new CountDownLatch(1);
+
+		CountDownLatch listenerLatch = new CountDownLatch(2);
+
+		@KafkaListener(id = GROUP_ID, topics = TOPIC_NAME, autoStartup = "false")
+		void listen(ConsumerRecord<Integer, String> ignored) {
+			listenerLatch.countDown();
+		}
+
+		@Bean
+		KafkaTemplate<Integer, String> template(ProducerFactory<Integer, String> producerFactory) {
+			return new KafkaTemplate<>(producerFactory);
+		}
+
+		@Bean
+		ProducerFactory<Integer, String> producerFactory() {
+			return new DefaultKafkaProducerFactory<>(KafkaTestUtils.producerProps(BROKER));
+		}
+
+		@Bean
+		ConcurrentKafkaListenerContainerFactory<Integer, String>
+		kafkaListenerContainerFactory(ConsumerFactory<Integer, String> consumerFactory) {
+
+			final ConcurrentKafkaListenerContainerFactory<Integer, String> factory =
+					new ConcurrentKafkaListenerContainerFactory<>();
+
+			factory.setConsumerFactory(consumerFactory);
+
+			factory.getContainerProperties().setConsumerRebalanceListener(new ConsumerAwareRebalanceListener() {
+				@Override
+				public void onPartitionsAssigned(Consumer<?, ?> consumer,
+												Collection<TopicPartition> partitions) {
+					partitionAssignedLatch.countDown();
+				}
+
+				@Override
+				public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
+					partitionRevokedLatch.countDown();
+				}
+			});
+			return factory;
+		}
+
+		@Bean
+		ConsumerFactory<Integer, String> consumerFactory() throws InterruptedException {
+			Properties propsDummy = new Properties();
+			propsDummy.setProperty(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, BytesDeserializer.class.getName());
+			propsDummy.setProperty(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+									StringDeserializer.class.getName());
+			propsDummy.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, BROKER);
+			propsDummy.setProperty(ConsumerConfig.GROUP_ID_CONFIG, "hello");
+			propsDummy.setProperty(ConsumerConfig.GROUP_PROTOCOL_CONFIG, "consumer");
+			propsDummy.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+			Consumer<Object, Object> dummyConsumer = new KafkaConsumer<>(propsDummy);
+
+			Thread.sleep(5000);
+			dummyConsumer.close();
+
+			final Map<String, Object> props = getConsumerProperties();
+			return new DefaultKafkaConsumerFactory<>(props);
+		}
+
+		@Bean
+		public ThreadPoolTaskExecutor threadPoolTaskExecutor() {
+			final ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+			executor.setCorePoolSize(5);
+			executor.setQueueCapacity(20);
+			executor.setMaxPoolSize(10);
+			return executor;
+		}
+
+		@Bean
+		public GenericContainer<?> brokerContainer() {
```

Comment:
I don't think this suppose to be a bean. 
See `@Container` and read its Javadocs how Testcontainers manage their lifecycle.

Reply:
- https://github.com/spring-projects/spring-kafka/pull/3237#discussion_r1596833689
- https://github.com/spring-projects/spring-kafka/pull/3237#discussion_r1596642357

I tried to do it. 
However, there is a major reason why i cannot use it.

1. As i commented on [this](https://github.com/spring-projects/spring-kafka/pull/3237#discussion_r1596833689), `broker` should load metadata before `consumer.subscribe()` is executed. 
2. If using `@Container`, `broker` container will be started successfully. however, there is no hooking point to execute `dummyConsumer` to make `broker` schedule load to `metadata` before `consumer` which are managed by `spring-kafka` execute `subscribe()`. 

We should make broker schedule to load metadata before `consumer` which is managed by `spring-kafka` are registered as spring bean. However, AFAIK, there is no annotation to define the execution order at the time of registering `Spring beans` on `@Configuration`.

I have this difficulty. Do you have good idea? 🙇‍♂️

Reply:
Yeah... That's too much involved.
I would give up already making this working 😄 .

As I said in other my comment, it might be a time to postpone such a sample until Apache Kafka has that coordinator fixed. 😢 

@sobychacko , WDYT?

Reply:
As you said, it seems to be better to postpone 😂😂.
Let me check that there is common way as last.
Because i'm not professional to kafka and still learning about it. 
Thus some mis-understanding can be existed. so, i will take another look to check mis-understanding made by me as last 😭

Reply:
Hey, @artembilan 
I found root cause of `kafka`, so tried to create a new PR on `apache/kafka`.
However it seems to have already been fixed by this PR (https://github.com/apache/kafka/pull/15698) 

The problem was caused by not retrying the error response (as opposed to the log say).
However, that PR is not included in `apache/kafka:3.7.0`.  😓 

Thus, we should postpone this PR. 
May I leave this PR in an open status and postpone it?

Thanks in advance!

Reply:
Yes. We can keep it opened until the next Apache Kafka `2.7.1`.
Thank you!